src/shared/mkfs-util.c

```diff
@@ -71,6 +71,11 @@ int make_filesystem(
                 mangled_label[sizeof(mangled_label)-1] = 0;
                 ascii_strupper(mangled_label);
 
+                /* mkfs.vfat: Labels with characters *?.,;:/\|+=<>[]" are not allowed */
+                for (char *p = mangled_label; *p; p++)
+                        if (strchr("*?.,;:/\\|+=<>[]\"", *p))
+                                *p = '_';
```

Comment:
I think it would make sense to split this whole mangling logic out:

```c
…
static const char* mangle_fat_label(const char *s, char ret[static 12]) {
        …
}
…
label = mangle_fat_label(s, mangled_label);
```

Reply:
Also, I figure non-ASCII chars and cntrol chars are not Ok either. Hence maybe list the chars that actually are OK?

Reply:
This turned out to be more involved than I though. I had to add some helpers to do this correctly, but I think we're now handling this much more robustly.